### ISSUE TITLE
waveforms + slip mode: issue with rolling loop + loop anchor == end

### ISSUE THREAD

**Issue by ronso0** [mixxxdj/mixxx#13243]
### Bug Description

activating a rolling loop with anchor = end while the deck is stopped results in duplicate / wrong loop and cue markers
![image](https://github.com/mixxxdj/mixxx/assets/5934199/b0e357be-5f80-4764-a0fb-6ce07db2e081)

Also, when the deck is playing, I find this view unexpected (loop_in marker in top waveform):
![image](https://github.com/mixxxdj/mixxx/assets/5934199/5f3cf001-b705-463c-96e3-d281bfdf3f4f)

### Version

2.5-alpha

**Comment by ronso0** [comment]
Did someone try to reproduce this?

Both the loop anchor option and the slip waveform are very nice features, but tbh I consider this conflict a blocker since it creates really confusing visuals (which kind of defeats the purpose IMHO).
@acolombier do you have time to take a look?

**Comment by acolombier** [comment]
I can try a take a look later. I know when there is a loop, the behaviour has become inconsistent, although it used to work fine, so I expect some of the marker rework is involved.

Could you please provide a detailed step to reproduce your usecase?

**Comment by ronso0** [comment]
I'm using rolling loops with the IN anchor. Just noticed this while the anchor was still set to END from a test routine.

I think that's it: just set the anchor to END and test with regular amd rolling loops.

**Comment by acolombier** [comment]
Might be fixed by #13422 
Could you please test it when you get the time?

**Comment by ronso0** [comment]
I merged 07113fdf314afdf09998f79d8d7571f041877c68 into 2.5
and it still looks like this, the original wave is offset by the loop size with LoopOut anchor.
![image](https://github.com/mixxxdj/mixxx/assets/5934199/cce6e72b-dec3-487e-9e1d-44bf7d78e699)

**Comment by acolombier** [comment]
I have managed to reproduce the issue, and I'm not sure whether this is expected behaviour or actually an issue. As it stands, the exit point is exclusive, so when you set the loop with anchor=exit, you immediately get back to the start point. 

Here is some workaround I can think of:

- Making the exit point inclusive, likely a breaking change
- Shift the loop by one frame forward when setting from exit, sounds like potentially disruptive 

Note that we could also not render certain markers in the slip waveform (e.g loop), although I personally like having them shown as a reference point.

**Comment by ronso0** [comment]
> I'm not sure whether this is expected behaviour or actually an issue. As it stands, the exit point is exclusive, so when you set the loop with anchor=exit, you immediately get back to the start point.

True, it is actually correct behaviour. I guess it was only the sudden jump that made me feel something's wrong. Sorry for the noise.

**Comment by acolombier** [comment]
No worries! :) 
Do you think we should do something about that? Potentially a `[NOTE]` in the doc?

**Comment by ronso0** [comment]
Yeah probably, I'll try the loop/slip stuff in real sessions and come back when I have ideas / proposals ready.

**Comment by daschuer** [comment]
What is the status here?

**Comment by ronso0** [comment]
Let's remove it from the 2.5.2 milestone.

**Comment by acolombier** [comment]
What is remaining to do here? I though this could be closed? The actual issue is fixed by #13422

**Comment by ronso0** [comment]
I didn't pay attention to the slip waveform lately when using rolling loops or reverse slip.
We can close this and reopen if someone notices an issue.
